src/runtime/generators.js

```diff
@@ -52,6 +52,31 @@
                       state);
   }
 
+  // The following unique object serves as a non-catchable exception raised by
+  // the implementation of Generator.prototype.return
+  var SENTINEL = new Object();
+  
+  function GeneratorWrapper(iterator) {
+    this.iterator = iterator;
+  }
+  GeneratorWrapper.prototype = {
+    next: function (v) {
+      return this.iterator.next(v);
+    },
+    throw: function (e) {
+      if (e === SENTINEL) {
+        if (this.iterator.return) {
+          this.iterator.return(e.value);
+        }
+        throw e;
```

Comment:
Otherwise, `g()` for a generator function `g` wouldn't be equivalent to `(function* wrap() { yield* g(); }())`...